mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessageZipper.java

```diff
@@ -49,9 +67,12 @@ public DeletedMessageZipper() {
 
     public void zip(DeletedMessageContentLoader contentLoader, Stream<DeletedMessage> deletedMessages, OutputStream outputStream) throws IOException {
         try (ZipArchiveOutputStream zipOutputStream = newZipArchiveOutputStream(outputStream)) {
-            ThrowingConsumer<DeletedMessage> putInZip = message -> putMessageToEntry(zipOutputStream, message, contentLoader);
```

Comment:
This code looks very complicated.
What about:
```
deletedMessages.forEach(message -> putMessageInEntry(zipOutputStream, message, contentLoader.load(message)));
```
And let putMessageToEntry handle InputStream emptyness?